src/proto.rs

```diff
@@ -1125,19 +1128,24 @@ impl Decoder for WebsocketProtocol {
         }
 
         // Bits 4-7
-        let opcode_value = fin_and_rsv & 31;
+        let opcode_value = fin_and_rsv & 0xF;
         let opcode = OpCode::try_from(opcode_value)?;
 
         if !fin && opcode.is_control() {
             return Err(Error::Protocol(ProtocolError::FragmentedControlFrame));
         }
 
+        // Bit 0
         let mask = payload_len_1 >> 7 != 0;
 
         if mask && self.role == Role::Client {
             return Err(Error::Protocol(ProtocolError::ServerMaskedData));
         }
 
+        if !mask && self.role == Role::Server {
+            return Err(Error::Protocol(ProtocolError::UnmaskedData));
+        }
```

Comment:
Perhaps we should make a `Relaxations` struct similar to `Limits` and default to all off. We could move `fail_fast_on_invalid_utf8` there too.

Reply:
Could that go in another PR? I like to not hold up the other commits because of this. Alternatively I could split off the commit that introduced this.

Reply:
Yeah that was the intent, I was just thinking loudly :)